cmd/kube-apiserver/app/server.go

```diff
@@ -612,7 +612,7 @@ func BuildAdmissionPluginInitializers(
 
 	admissionPostStartHook := func(context genericapiserver.PostStartHookContext) error {
 		discoveryRESTMapper.Reset()
-		go utilwait.Until(discoveryRESTMapper.Reset, 30*time.Second, context.StopCh)
+		go utilwait.Until(func() { discoveryRESTMapper.Reset() }, 30*time.Second, context.StopCh)
```

Comment:
What is the purpose of the added func() wrapper?

Reply:
I see, you added an error return. Why is that ignored here?

Reply:
Well, formerly it was logged.

I'd also argue that dropping a return value completely like this makes the code hard to understand, as readers won't know that an error was possible and also will not be sure if dropping it was deliberate or not (they'll probably assume it was accidental).

Reply:
@lavalamp I added back the error logging and added TODOs, see https://github.com/kubernetes/kubernetes/pull/68859/commits/1041fab2183c8d48dc7e2012319de239100a90ec.

I will change the signature of `CachedDiscoveryInterface.Invalidate` back to not reporting error after solving #68865. Changing the interface back and forth is hostile to client-go users. So I can drop this PR is we can live with the flaky tests in #68735 for a few more weeks..

Reply:
> Changing the interface back and forth is hostile to client-go users. So I can drop this PR is we can live with the flaky tests in #68735 for a few more weeks..

That seems good to me. I'd like to see us land on a coherent interface with implementations that match the documentation rather than flip flop the interface rapidly